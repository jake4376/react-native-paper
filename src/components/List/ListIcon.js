/* @flow */

import color from 'color';
import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Icon, { type IconSource } from '../Icon';
import { withTheme } from '../../core/theming';
import type { Theme } from '../../types';

type Props = {
  /**
   * Icon to show.
   */
  icon: IconSource,
  /**
   * @optional
   */
  theme: Theme,
  style?: any,
};

/**
 * A component to show an icon in a list item.
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { ListSection } from 'react-native-paper';
 *
 * const MyComponent = () => (
 *   <ListSection.Icon icon="folder" />
 * );
 * ```
 */
class ListIcon extends React.Component<Props> {
  static displayName = 'ListSection.Icon';

  render() {
    const { icon, theme, style } = this.props;

    const iconColor = color(theme.colors.text)
      .alpha(0.54)
      .rgb()
      .string();

    return (
      <View style={[styles.item, style]} pointerEvents="box-none">
        <Icon source={icon} size={24} color={iconColor} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  item: {
    margin: 8,
    height: 40,
    width: 40,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default withTheme(ListIcon);
