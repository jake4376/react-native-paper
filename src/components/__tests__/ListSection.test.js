/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import ListSection from '../List/ListSection';
import ListItem from '../List/ListItem';
import ListIcon from '../List/ListIcon';
import ListSubheader from '../List/ListSubheader';

it('renders list section without subheader', () => {
  const tree = renderer
    .create(
      <ListSection>
        <ListItem
          title="First Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
        <ListItem
          title="Second Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list section with subheader', () => {
  const tree = renderer
    .create(
      <ListSection>
        <ListSubheader>Some title</ListSubheader>
        <ListItem
          title="First Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
        <ListItem
          title="Second Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
