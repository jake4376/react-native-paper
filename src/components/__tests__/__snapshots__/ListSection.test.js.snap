// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list section with subheader 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "animation": Object {
        "scale": 1,
      },
      "colors": Object {
        "accent": "#03dac4",
        "backdrop": "rgba(0, 0, 0, 0.5)",
        "background": "#f6f6f6",
        "disabled": "rgba(0, 0, 0, 0.26)",
        "error": "#B00020",
        "notification": "#f50057",
        "placeholder": "rgba(0, 0, 0, 0.54)",
        "primary": "#6200ee",
        "surface": "#ffffff",
        "text": "#000000",
      },
      "dark": false,
      "fonts": Object {
        "light": "HelveticaNeue-Light",
        "medium": "HelveticaNeue-Medium",
        "regular": "Helvetica Neue",
        "thin": "HelveticaNeue-Thin",
      },
      "roundness": 4,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 13,
          },
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontWeight": "500",
          },
          undefined,
        ],
      ]
    }
  >
    Some title
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list section without subheader 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "animation": Object {
        "scale": 1,
      },
      "colors": Object {
        "accent": "#03dac4",
        "backdrop": "rgba(0, 0, 0, 0.5)",
        "background": "#f6f6f6",
        "disabled": "rgba(0, 0, 0, 0.26)",
        "error": "#B00020",
        "notification": "#f50057",
        "placeholder": "rgba(0, 0, 0, 0.54)",
        "primary": "#6200ee",
        "surface": "#ffffff",
        "text": "#000000",
      },
      "dark": false,
      "fonts": Object {
        "light": "HelveticaNeue-Light",
        "medium": "HelveticaNeue-Medium",
        "regular": "Helvetica Neue",
        "thin": "HelveticaNeue-Thin",
      },
      "roundness": 4,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;
